### DISCUSSION TITLE
lightning-flash Multilabel

### DISCUSSION THREAD

**Discussion by jorgeutd** [PyTorchLightning/lightning-flash#395] [Q&A]
Hello lightning-flash team,

Does flash support multilabel text classification? I tried a couple of times with the target column in multiples formats and it did not work.

Thanks,

Jorge

**Comment by ethanwharris** [comment]
Hi @jorgeutd thanks for the question! This looks like a bug as it was expected to work. I've opened a couple of issues:

- #397 A bug issue to get multilabel text working properly
- #396 A feature issue to add an example showing multilabel text classification in action

I'll get back to you here once these are done. Hope that helps :smiley:

**Comment by ethanwharris** [comment]
Hi @jorgeutd - as of #401 this is now supported :smiley: Take a look at the docs here: https://lightning-flash.readthedocs.io/en/latest/reference/text_classification_multi_label.html

**Reply by jorgeutd** [reply]
Thank you Ethan. I am having hard time making this work.  
I am passing the data this way: 
![image](https://user-images.githubusercontent.com/30812821/121778718-2bfc7480-cb66-11eb-81b0-1dcde4345bb2.png)

I also tried to reprouce the toxic comment example and I am getting the zip error: 
![image](https://user-images.githubusercontent.com/30812821/121778738-46cee900-cb66-11eb-83ba-b7ef7c57e02b.png)

Thanks,

Jorge

**Reply by ethanwharris** [reply]
Hi Jorge,

Thanks for sticking with it :smiley: The data format we used for the toxic comments problem looks like this:
```
text, class_1, class_2, class_3, ...
"", 1, 0, 0, ...
```
Is that an option for you? We could add support for a single column with a list of labels, but don't have it currently. Regarding the zip error, my guess id it is some OS incompatibility. Can I ask what operating system you are using?

**Reply by jorgeutd** [reply]
Hi Ethan,

Thank you again for your quick reply and for helping on this. I am running this in Colab initially just to test that I can make it work with Flash.

I changed the labels format to the one that you advised and also I was able to unzip the toxic comment data. In both cases the datamodule is getting stuck in some labels issues, please see the two images below with the same error for both datasets:

![image](https://user-images.githubusercontent.com/30812821/121806777-3e86b480-cc1f-11eb-8485-4c0fc19edfd6.png)

I am saving my train / test as csv:

![image](https://user-images.githubusercontent.com/30812821/121806805-5bbb8300-cc1f-11eb-88d3-7db41b334701.png)


![image](https://user-images.githubusercontent.com/30812821/121806893-bf45b080-cc1f-11eb-8698-dee699ee45e0.png)


It could definitely be a csv related problem, but it is happening in both datasets and I did not changed anything on the toxic comment files just downloaded. 


Thanks,

Jorge

**Reply by ethanwharris** [reply]
Hi Jorge,

Can I ask what version of Flash you are using? This support is only available on master at the moment, so you would need to install with:
```
pip install git+https://github.com/PytorchLightning/lightning-flash.git@master#egg=lightning-flash[text]
```

Hope that helps :smiley:

**Reply by jorgeutd** [reply]
Ethan,

This was the issue. I am now able to reproduce the example with any issues. Also, I was looking into the flash doc and I saw that for a summarization tasks the target_max_length argument is available, do you have anything like this for a classification tasks or what is the default max_length for the input text 512?

I am getting the following error now for my custom data:
![image](https://user-images.githubusercontent.com/30812821/121809753-0fc30b00-cc2c-11eb-8e65-d554f459e6ab.png)

Has to be the way the csv is encoding the text input field. What would you recommend here:

Again, thank you so much Ethan.

**Reply by jorgeutd** [reply]
Ethan,

Is there another option to pass the data into the dataloader for this type of tasks? Where I could find examples?

Thanks,

Jorge

**Reply by jorgeutd** [reply]
Ethan,

Everything is working now. Thank you so much.

**Reply by ethanwharris** [reply]
@jorgeutd Awesome! Glad you got it working. Please don't hesitate to reach out again if you have any feedback :smiley:
